### DISCUSSION TITLE
Difficulties setting up own tap with modified formula

### DISCUSSION THREAD

**Discussion by SwishSwushPow** [Homebrew/discussions#4347] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.6
ORIGIN: https://github.com/Homebrew/brew
HEAD: 0c1031f2fd71016f0f5df699caaed384487bfb16
Last commit: 12 days ago
Core tap JSON: 22 Mar 11:02 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_blizzard_avalanche
Clang: 14.0.0 build 1400
Git: 2.37.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo giga/tap) checkout
```

### Description of issue

Hi everyone 👋 

For work I need to use a different version of `gdal`. Currently the latest version available via `homebrew/core` is 3.6.3 yet I need some version of 3.5.x (e.g. 3.5.3_2). I would like to have the 3.6.x version symlinked and install the 3.5.x version in parallel as keg only.

I tried to create a new tap (via `brew tap-new giga/homebrew-tap`) and add the formula for `gdal` 3.5.3_2 there (I took the file from https://github.com/Homebrew/homebrew-core/blob/dddd729ecba04dcf2a25f50e2143ded3b44cd13f/Formula/gdal.rb). I renamed it from `gdal.rb` to `gdal@3.5.3.rb` and changed the name within the file to `GdalAT353` as to dodge any naming conflicts. I also added `keg_only` to the formula. I have not pushed anything to a remote repository yet.

Now I run into two issues. `brew audit --new gdal@3.5.3 gives
```
% brew audit --new gdal@3.5.3 --tap giga/tap --verbose
giga/tap/gdal-3.5.3:
  * New formulae should not define a revision.
```
I'm not sure what to do with this. Yes I know that it is an older version of an existing formula, but that is why I am trying to add a new one with a different name.

If I ignore this and try to install it with `brew install giga/tap/gdal@3.5.3` it is not able to find the manifest_bottle
```
==> Fetching giga/tap/gdal-3.5.3
==> Downloading https://ghcr.io/v2/homebrew/core/gdal-3.5.3/manifests/3.5.3_2
curl: (22) The requested URL returned error: 404
```
I assume this happens because the manifest only exists for the normal `gdal` formula. At this point I am not sure how to continue or if my goal is even achievable. I would be very happy about any pointers where to go from here.

I tried to follow https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap as best as possible.

Let me know if I can provide any more information.

Best regards,
Björn

Edit: The reason why I try to do this with a new tap is that I don't believe the versioned formula would fit the posted `Acceptable versioned formulae` guidelines (https://docs.brew.sh/Versions).

**Comment by gromgit** [comment, marked as answer]
Instead of copying manually from the repo, you should've done this:
```
brew extract --version=3.5.3 gdal giga/tap
```
That would do all the right things, including removing the `bottle` block that's probably causing your manifest errors.

**Reply by SwishSwushPow** [reply]
Ah thank you for that quick reply. I had to `brew tap homebrew/core` but afterwards I was able to extract the formula (which, as you mentioned, basically just removed the `bottle` block. So I wasn't far off I suppose). Now I have to sort out a build error, but hopefully I'll be able to figure this out on my own.

Thanks again!
